### ISSUE TITLE
run_experiment in BasicAer fails after optmization_level=3

### ISSUE THREAD

**Issue by 1ucian0** [Qiskit/qiskit-terra#2461]
### Information

- **Qiskit Terra version**: master

### What is the current behavior?

```python
backend = BasicAer.get_backend('qasm_simulator')
qr = QuantumRegister(3, 'q')
cr = ClassicalRegister(3, 'c')
circuit = QuantumCircuit(qr, cr)
circuit.h(qr[1])
circuit.cx(qr[1], qr[2])
circuit.measure(qr, cr)

job = execute(circuit, backend, optimization_level=3)
print(job.result().get_counts())
```
```
qiskit.providers.basicaer.exceptions.BasicAerError: 'qasm_simulator encountered unrecognized operation "unitary"'
```

**Comment by 1ucian0** [comment]
other optimization levels work as expected.

**Comment by kdk** [comment]
Should be resolved by #2342.

**Comment by 1ucian0** [comment]
indeed.
